### PR TITLE
Localize tools.

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

File: src/vcpkg/tools.cpp [hunk 1]
```diff
@@ -174,10 +173,9 @@ namespace vcpkg
             return {std::move(output), expected_left_tag};
         }
 
-        auto error_prefix =
-            msg::format_error(msgUnexpectedToolOutput, msg::tool_name = tool_name, msg::path = exe_path).extract_data();
-        error_prefix.push_back('\n');
-        return {std::move(error_prefix) + std::move(output), expected_right_tag};
+        return std::move(msg::format_error(msgUnexpectedToolOutput, msg::tool_name = tool_name, msg::path = exe_path)
```
Comment (ras0219-msft) [inline src/vcpkg/tools.cpp:176]: Why does this all need to be wrapped in a `std::move()` -- isn't it already an rvalue?

File: src/vcpkg/tools.cpp [hunk 2]
```diff
@@ -174,10 +173,9 @@ namespace vcpkg
...
+                             .append_raw('\n')
+                             .append_raw(std::move(output)));
```
Comment (ras0219-msft) [inline src/vcpkg/tools.cpp:178]: I think this move does nothing

File: src/vcpkg/tools.cpp [hunk 3]
```diff
@@ -306,10 +304,8 @@ namespace vcpkg
             cmd.string_arg(exe_path);
             return run_to_extract_version(Tools::NUGET, exe_path, std::move(cmd))
 #if !defined(_WIN32)
-                .map_error([](std::string&& error) {
-                    error.push_back('\n');
-                    error.append(msg::format(msgMonoInstructions).extract_data());
-                    return std::move(error);
+                .map_error([](LocalizedString&& error) {
+                    return std::move(error.append_raw('\n').append(msg::format(msgMonoInstructions)));
```
Comment (ras0219-msft) [inline src/vcpkg/tools.cpp:308]: I think this move needs to be rescoped:
```suggestion
                    return std::move(error).append_raw('\n').append(msgMonoInstructions);
```

**Review by ras0219-msft — APPROVED** [review]

File: src/vcpkg/tools.cpp (hunk 1 above)
Comment (BillyONeal) [reply src/vcpkg/tools.cpp:176]: `LocalizedString::append` returns `LocalizedString&`, not `LocalizedString&&`.

**Pull request merged**